File: 프로그래머스/0/181865. 간단한 식 계산하기/간단한 식 계산하기.cpp
#include <string>
#include <vector>
#include <sstream>

using namespace std;

int calculate(const std::string& expression) {
    std::istringstream iss(expression);
    int result;
    char op;

    iss >> result;

    while (iss >> op) {
        int operand;
        iss >> operand;

        if (op == '+') {
            result += operand;
        } else if (op == '-') {
            result -= operand;
        }else if (op == '*') {
            result *= operand;
        }
    }

    return result;
}
int solution(string binomial) {
    std::istringstream iss(binomial);
    std::string part;
    int result = 0;

    while (std::getline(iss, part, '+')) {
        result += calculate(part);
    }

    return result;
}

File: 프로그래머스/0/181865. 간단한 식 계산하기/간단한 식 계산하기.py
def solution(binomial):
    answer = 0
    binomial=binomial.split(' ')
    a,op,b=binomial
    
    if op=='+':
        answer=int(a)+int(b)
    elif op=='-':
        answer=int(a)-int(b)
    else:
        answer=int(a)*int(b)
    return answer